pkg/controller/controller.go

```diff
@@ -235,3 +241,46 @@ func sendMessage(c *config.Config, msg string) {
 		go notifier.SendMessage(msg)
 	}
 }
+
+func configWatcher(c *config.Config) {
+	configPath := os.Getenv("CONFIG_PATH")
+	configFile := filepath.Join(configPath, config.ConfigFileName)
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Logger.Fatal("Failed to create file watcher ", err)
+	}
+	defer watcher.Close()
+
+	done := make(chan bool)
+	go func() {
+		for {
+			select {
+			case _, ok := <-watcher.Events:
```

Comment:
Do we need to check type of events?
What kind of events we can get on channel ?

Reply:
These are file change events (not k8s events). Whenever file gets updated, we get event and we restart the pod.